src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCuiCallbackHandler.java

```diff
@@ -55,10 +59,19 @@ public List<CaseEvent> handledEvents() {
     private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
         CaseData updatedData = getUpdatedCaseData(callbackParams);
 
-        return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(updatedData.toMap(objectMapper))
-            .state(CaseState.AWAITING_APPLICANT_INTENTION.name())
-            .build();
+        AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder responseBuilder =
+            AboutToStartOrSubmitCallbackResponse.builder().data(updatedData.toMap(objectMapper));
+
+        String responseLanguage = Optional.ofNullable(updatedData.getCaseDataLiP())
+            .map(CaseDataLiP::getRespondent1LiPResponse)
+            .map(RespondentLiPResponse::getRespondent1ResponseLanguage)
+            .orElse(null);
+
+        if (responseLanguage != null  && !Language.BOTH.toString().equals(responseLanguage)) {
+            responseBuilder.state(CaseState.AWAITING_APPLICANT_INTENTION.name());
```

Comment:
I think this will break the existing submission. what if there is no responseLanguage? shouldn't it continue as normal? 1. I would put the logic of checking whether this is both inside response lip (so that we can reuse it elsewhere not only in this handler) - then you can also put a unit test just for that condition. 2. i would continue as normal if responseLanguage is null. but will not change the claim state if language is both

Reply:
I got your point. I have fixed it now. Please have a look.